From 4df8d2b44cde1b8fa26ddebc6e90468dd8403faa Mon Sep 17 00:00:00 2001
From: Cai Song <cais.fnst@cn.fujitsu.com>
Date: Tue, 1 Aug 2017 06:16:58 -0400
Subject: [PATCH] net.rpc_tests/*: add path of LTP to environment

add path of LTP to environment firstly when executing remote
command, which help to locate the remote command on the target
machine.

Signed-off-by: Cai Song <cais.fnst@cn.fujitsu.com>
---
 testcases/lib/test_net.sh | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/testcases/lib/test_net.sh b/testcases/lib/test_net.sh
index 10cf6da..cb71936 100644
--- a/testcases/lib/test_net.sh
+++ b/testcases/lib/test_net.sh
@@ -89,6 +89,11 @@ tst_rhost_run()
 		return 1
 	fi
 
+	if [ -z "$LTPROOT" ]; then
+		LTPROOT="/opt/ltp"
+	fi
+	pre_cmd='export PATH=$PATH:'$LTPROOT'/testcases/bin && '$pre_cmd
+
 	local output=
 	local ret=0
 	if [ -n "${TST_USE_SSH:-}" ]; then
@@ -277,7 +282,7 @@ tst_add_ipaddr()
 
 	if [ $type = "lhost" ]; then
 		tst_resm TINFO "set local addr $(tst_ipaddr)/$mask"
-		ip addr add $(tst_ipaddr)/$mask dev $iface
+		ip addr show $iface | grep -nqE $(tst_ipaddr) || ip addr add $(tst_ipaddr)/$mask dev $iface
 		return $?
 	fi
 
-- 
2.7.4

